src/content/en/fundamentals/performance/resource-prioritization.md

```diff
@@ -0,0 +1,351 @@
+
+Note that the use of `crossorigin` here is important; without this attribute,
+the preloaded font will be ignored by the browser, and a new fetch will take
+place. This is because fonts are expected to be fetched anonymously by the
+browser, and the preload request is only made anonymous by using the
+`crossorigin` attribute.
+
+Caution: If you’re using a CDN, such as Google Fonts, be sure that the font files
+you’re preloading match the ones in the CSS, which can be tricky due to unicode
+ranges, weights, and font variants. Fonts can also be regularly updated, and if
+you’re preloading an old version while using the CSS for a newer one, you may
+end up downloading two versions of the same font and wasting your users’
+bandwidth. Consider using `<link rel="preconnect">` instead for easier
+maintenance.
+
+### Use-case: Critical Path CSS and JavaScript
+
+When talking about page performance, one useful concept is that of the
+“critical path”, most often applied to CSS. This refers to the resources that
+need to be loaded before your initial render, and are thus critical to getting
+the first pixels to the user’s screen.
+
+Previously, the recommendation was to inline this content into your HTML.
+However, in a multi-page, server-side rendered scenario, this quickly grows into
+a lot of wasted bytes. It also makes versioning harder, as any change in the
+critical code invalidates any page that has it inlined.
+
+`<link rel="preload">`, on the other hand, allows you to keep the benefits of
+individual file versioning and caching, while giving you mechanism to request
+the resource as soon as possible.
+
+    <link rel="preload" as="script" href="super-important.js">
+    <link rel="preload" as="style" href="critical.css">
+
+There’s a downside compared to inlining, however: with `<link rel="preload">`
+you’re still subject to an extra roundtrip. This extra roundtrip comes from the
+fact that the browser first has to fetch the HTML, and only then does it find
+out about the next resources.
+
+One way around the extra roundtrip is to use
+[HTTP/2](/web/fundamentals/performance/http2/#server_push)
+push instead, where you preemptively attach the critical assets to the same
+connection through which you’re sending the HTML. This way, you guarantee that
+there’s no downtime between the user’s browser retrieving the HTML and starting
+the download of the critical assets. Be mindful when using HTTP/2 push, though,
+as it’s a very forceful way of controlling the user’s bandwidth usage (“server
+knows best”), and leaves the browser very little room for making its own
+decisions, such as not retrieving a file that is already in its cache!
+
+## Preconnect
+
+`<link rel="preconnect">` informs the browser that your page intends to
+establish a connection to another origin, and that you’d like the process to
+start as soon as possible.
+
+Establishing connections often involves significant time in slow networks,
```

Comment:
I think it's when it comes... rather than where it comes.

Reply:
Done.